qiita_db/sql_connection.py

```diff
@@ -460,176 +450,199 @@ def execute_fetchall(self, sql, sql_args=None):
 
         return result
 
-    def _check_queue_exists(self, queue_name):
-        """Checks if queue `queue_name` exists in the handler
 
-        Parameters
-        ----------
-        queue_name : str
-            The name of the queue
+class Transaction(object):
+    """Encapsulates a DB transaction
 
-        Returns
-        -------
-        bool
-            True if queue `queue_name` exist in the handler. False otherwise.
-        """
-        return queue_name in self.queues
+    A transaction is defined by a series of consecutive queries that need to
+    be applied to the database as a single block.
 
-    def create_queue(self, queue_name):
-        """Add a new queue to the connection
+    Parameters
+    ----------
+    name : str
+        Name of the transaction.
+    """
 
-        Parameters
-        ----------
-        queue_name : str
-            Name of the new queue
+    _regex = re.compile("^{(\d+):(\d+):(\d+)}$")
+
+    def __init__(self, name):
+        # The name is useful for debugging, since we can identify the
+        # failed queue in errors
+        self._name = name
+        self._queries = []
+        self._results = []
+        self._index = 0
+        self._conn_handler = SQLConnectionHandler()
+
+    def _raise_execution_error(self, sql, sql_args, error):
+        """Rollbacks the current transaction and raises a useful error
+
+        The error message contains the name of the transaction, the failed
+        query, the arguments of the failed query and the error generated.
 
         Raises
         ------
-        KeyError
-            Queue name already exists
+        ValueError
         """
-        if self._check_queue_exists(queue_name):
-            raise KeyError("Queue %s already exists" % queue_name)
+        self.rollback()
+        raise ValueError(
+            "Error running SQL query in transaction %s:\n"
+            "Query: %s\nArguments: %s\nError: %s\n"
+            % (self._name, sql, str(sql_args), error))
```

Comment:
Might be a good idea to do `str(error)` since the places where this is being used pass an exception object.